src/components/AddressForm.tsx

```diff
@@ -0,0 +1,239 @@
+import {
+	InputAdornment,
+	TextField,
+	Typography,
+	Button,
+	Select,
+	MenuItem,
+	SelectChangeEvent,
+	InputLabel,
+	FormControl,
+	FormHelperText,
+} from '@mui/material'
+import 'react-toastify/dist/ReactToastify.css'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { FaAngleRight } from 'react-icons/fa6'
+import { useState, useEffect } from 'react'
+import { toast } from 'react-toastify'
+import { statesOfUSA } from '~/data/statesOfUSA'
+import { SettingFormsProps } from '~/lib/types'
+
+const addressSchema = z.object({
+	address: z.string().min(1, { message: 'Address cannot be empty.' }),
+	addressOptional: z.string().optional(),
+	city: z.string().min(1, { message: 'City cannot be empty.' }),
+	state: z.string().min(1, { message: 'State cannot be empty.' }),
+	zipCode: z.string().min(1, { message: 'Zip code cannot be empty.' }),
+})
```

Comment:
I'd move the schemas to a separate space (maybe a new package?), because they should be reused by the backend as well.

Reply:
resolved in account-settings branch